lib/galaxy/datatypes/media.py

```diff
@@ -203,12 +175,13 @@ def set_meta(self, dataset, overwrite=True, **kwd):
         """Set the metadata for this dataset from the file contents
         """
 
-        fd = wave.open(dataset.dataset.file_name, 'rb')
-        dataset.metadata.rate = fd.getframerate()
-        dataset.metadata.nframes = fd.getnframes()
-        dataset.metadata.sampwidth = fd.getsampwidth()
-        dataset.metadata.nchannels = fd.getnchannels()
-        fd.close()
-
-
-Binary.register_sniffable_binary_format('wav', 'wav', WAV)
+        try:
+            fd = wave.open(dataset.dataset.file_name, 'rb')
+            dataset.metadata.rate = fd.getframerate()
+            dataset.metadata.nframes = fd.getnframes()
+            dataset.metadata.sampwidth = fd.getsampwidth()
+            dataset.metadata.nchannels = fd.getnchannels()
+        except wave.Error:
+            pass
```

Comment:
This means it'll not fail setting metadata even if wave.Error is raised, is that intentional ? (which is fine, the metadata elements are marked as optional, just checking since it wasn't in the previous code.)

Reply:
Yeah, if it's been successfully sniffed but metadata setting fails for some reason? That seems reasonable?

Reply:
Sure, I guess it'll have an impact on how you write tools. If it's optional the tool author should probably check if the attributes are set if there is some logic using these in the tool. Either way is fine with me.